conanfile.py

```diff
@@ -317,13 +343,35 @@ def build_configure(self):
                 args.append('--enable-videotoolbox' if self.options.videotoolbox else '--disable-videotoolbox')
                 args.append('--enable-securetransport' if self.options.securetransport else '--disable-securetransport')
 
-            if self.settings.os == "Windows":
+            if self.settings.os == "Windows" and not self._use_mingw:
                 args.append('--enable-libmfx' if self.options.qsv else '--disable-libmfx')
 
+            if tools.cross_building(self.settings):
+                args.append('--enable-cross-compile')
+                args.append('--target-os=%s' % self._target_os)
+                cc = os.environ["CC"]
+                cxx = os.environ["CXX"]
+                args.append('--cc=%s' % cc)
+                args.append('--cxx=%s' % cxx)
+                arch = {'armv7': 'arm',
```

Comment:
raise an error here if arch is not supported (otherwise, consumer will have an obscure error because of `---arch=None`)
probably, better to move to its own function (like `_target_os`)
also, check pls if other architectures are supported by ffmpeg (ppc, mips, sparc)